packages/tools/src/utilities/math/polyline/isPointInsidePolyline.ts

```diff
@@ -0,0 +1,98 @@
+import type { Types } from '@cornerstonejs/core';
+
+/**
+ * Determines whether a 2D point is inside a polyline.
+ * The algorithm works by drawing horizontal rays from the point to the right side
+ * of the polygon, counting the number of times the ray intersects a polygon edge.
+ *  If the winding number is odd, the point is inside the polygon.
+ *
+ * @param point - The 2D point to check.
+ * @param polyline - The polyline represented as an array of 2D points.
+ * @returns A boolean indicating whether the point is inside the polyline.
+ */
+export function isPointInsidePolyline2D(
+  point: Types.Point2,
+  polyline: Types.Point2[]
+) {
+  const n = polyline.length;
+  let windingNumber = 0;
+  const x = point[0],
+    y = point[1];
+
+  for (let i = 0; i < n; i++) {
+    const j = (i + 1) % n;
```

Comment:
How about const iNext or something desriptive - I think of j as another iteration index.